GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/emitters/GVREmitter.java

```diff
@@ -0,0 +1,370 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.particlesystem.emitters;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.particlesystem.Particles;
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+import java.util.ArrayList;
+
+
+/**
+ * The base emitter class from which shape-specific emitter classes inherit.
+ * This class is used to set up the the behaviour of the particle system in general.
+ * Also, this is used to set the specific particle properties.
+ *
+ * Every emitter has a set of scene objects as its children. These scene
+ * objects each have a mesh attached to them, with the number of vertices in the mesh
+ * being equal to the emit rate of the emitter, i.e., every second a new scene object with a mesh
+ * having an emitRate number of vertices is added to the emitter as a child.
+ * These vertices act as the particles of the system. Consequently, all those
+ * children objects which have exceeded their age limits are deleted every second.
+ *
+ */
+
+public class GVREmitter extends GVRSceneObject implements GVRDrawFrameListener {
```

Comment:
All the other scene objects (GVRCameraSceneObject, GVRWebViewSceneObject, GVRTextViewSceneObject, GVRVVideoSceneObject) implement it this way. They do:

gvrContext.registerDrawFrameListener(this);

and then implement the onDrawFrame function. Are you suggesting that we change the way we do onDrawFrame everywhere? I thought that registering the scene object as the draw frame listener was the right way to get GearVRF to call onDrawFrame. Because SceneObject implements DrawFrameListener, it already has a public onDrawFrame function.